### DISCUSSION TITLE
pdf plumb returning "none"

### DISCUSSION THREAD

**Discussion by GrantMWu** [jsvine/pdfplumber#450] [Ask for help with specific PDFs]
I'm trying to use pdfplumber to extract text from a pdf, but I'm getting a return of "none" for certain pages. For other pages, the below code works fine. I suspect this has something to do with the way the pdf is set up, and I'm wondering if there is an easy work around. my code is below and sample pdf is attached

import pdfplumber

with pdfplumber.open(test_pdf) as pdf:
    page = pdf.pages[0]
    text = page.extract_text()

print(text)
[test_pdf.pdf](https://github.com/jsvine/pdfplumber/files/6632151/test_pdf.pdf)

I'm pretty new to coding, outside of a few python classes in college.

**Comment by samkit-jain** [comment]
Hi @gmwu843 Appreciate your interest in the library and wish you well in your Python journey. When dealing with text extraction related issues, the first step would be to check if `pdfminer.six` is able to extract it or not. Behind the scenes, `pdfplumber` relies on [pdfminer.six](https://github.com/pdfminer/pdfminer.six). They also provide a handy tool for text extraction and can be found [here](https://github.com/pdfminer/pdfminer.six/blob/develop/tools/pdf2txt.py). When you run the PDF on it like `python pdf2txt.py test_pdf.pdf`, you'll notice that it prints nothing because it is unable to read any text which also explains why `pdfplumber` is not able to read as well.

The reason _could_ be that font information/mapping/cmap is missing in the PDF. When viewing in a PDF reader, the text is copyable because the reader might be substituting the missing mappings with a default font.

In order to extract the text correctly, you can repair the PDF using [Ghostscript](https://www.ghostscript.com/index.html) like so
```sh
gs -o output.pdf -sDEVICE=pdfwrite input.pdf
```

When using the repaired PDF, you'll be able to extract the text properly. Attaching the repaired PDF [here](https://github.com/jsvine/pdfplumber/files/6633332/output.pdf) for your reference.

**Reply by GrantMWu** [reply]
Thanks! That worked. Appreciate the help.
